features/FEATURE_LWIP/lwip-interface/lwip_stack.c

```diff
@@ -104,7 +104,7 @@ static void mbed_lwip_socket_callback(struct netconn *nc, enum netconn_evt eh, u
 /* TCP/IP and Network Interface Initialisation */
 static struct netif lwip_netif;
 static bool lwip_dhcp = false;
-static char lwip_mac_address[NSAPI_MAC_SIZE] = "\0";
+static char lwip_mac_address[NSAPI_MAC_SIZE + 1] = "\0";
```

Comment:
+1 is not required - NSAPI_MAC_SIZE appears to already be sized to include the terminator (18). snprintf itself also can just be passed NSAPI_MAC_SIZE, which means it will write a maximum of 17 characters plus a terminator.

So this patch overcompensates - it was an 18-byte buffer with a 19-byte snprintf - this version has it as a 19-byte buffer with an 18-byte printf.
 
It actually seems wrong that NSAPI_MAC_SIZE even exists at the generic layer. It's up to any individual interface how long a MAC address can be. NanostackInterface returns 23-character MAC addresses.